design/baremetal-operator/bulk-set-bios-config.md

```diff
@@ -0,0 +1,228 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# bulk-set-bios-config
+
+## Status
+
+implementable
+
+## Summary
+
+Provide the ability to set the same BIOS Configuration across multiple similar
+servers from the same vendor. Retrieve the current BIOS Configuration settings
+from the host via Ironic and use that as a template. Allow a user to set new
+values for the BIOS Configuration which can be copied to multiple hosts of the
+same type. Validate the values before sending to Ironic via the clean-steps API.
+
+## Motivation
+
+When provisioning a large amount of machines with identical settings, our
+customers want to provide a known, validated payload with all the desired BIOS
+configuration. This ensures all the machines will have the identical settings,
+reducing issues caused by unexpected configuration differences.
+
+The challenge that BIOS configuration presents is that no two hardware vendors
+use the same names for settings that control the same behavior, and the
+available settings often overlap but have significant differences. There
+currently is an approach that would abstract these differences for individual
+vendors
+[individual set](https://github.com/metal3-io/baremetal-operator/pull/302).
+This new proposal would not require Metal3 to
+contain the vendor specific names, it will be entirely transparent and
+inherently support all vendors. This new proposal could also work in
+conjunction with the individual settings approach. A user could initially set
+up a machine using that method and then push the configuration to multiple
+machines as described here.
+
+Most hardware vendors support a way to apply bulk settings, often by manually
+configuring one host, exporting those settings in a vendor-specific file
+format, and then importing that file on the BMCs of other hosts. Accepting that
+the format for these profiles is vendor-specific eliminates a lot of the
+implementation complexity, at the risk of moving that complexity to the
+end-user. This requirement is similar to something Dell has proposed in the
+Ironic community
+([hw config](https://review.opendev.org/c/openstack/ironic-specs/+/740721))
+to export and import an entire system iDRAC inventory including RAID and
+networking requirements. Compared to that approach, this proposal attempts
+a more lightweight, vendor-neutral approach just for BIOS configuration.
```

Comment:
Yes this could be clearer. It really is not vendor-neutral per-se but it uses the vendor specific names and doesn't require a mapping layer that would convert the names to a common set of names. To transfer the settings to other hosts using k8s commands, only servers from the same vendors can be used as the names are specific to that vendor.  I will rewrite this.